app/Services/SparqlService.php

```diff
@@ -74,148 +150,123 @@ public function performSparqlQuery($query, $method = 'GET', $format = 'turtle')
+     * @param  string $query
+     * @return string
+     */
+    public static function transformQuery($query)
+    {
+        $query = str_replace('%23', '#', $query);
+        $query = urlencode($query);
+        $query = str_replace('+', '%20', $query);
 
-        return $response;
+        return $query;
     }
 
     /**
-     * Make and return the request URI based on
-     * the passed query and the configured endpoint
+     * Perform the query to the SPARQL endpoint
+     *
+     * Standard use auht. This is not needed for the GET SELECT,
+     * but to make extra logic for a "less safe exception" is ... not logic
      *
+     * 'application/sparql-query' is set es Content-type
+     *
+     * @param  string $verb    HTTP verb GET / POST
      * @param  string $query
-     * @param  string $format
-     * @return string
+     * @param  array  $options
+     * @return Psr\Http\Message\StreamInterface
      */
-    private function makeGetRequestUri($query, $format = null)
+    private function executeQuery($verb, $query, $options = [])
     {
-        $requestUri = $this->endpoint . '?query=' . $query;
+        $options['auth'] = [$this->username, $this->password, 'digest'];
+        $options['Content-Type'] = 'application/sparql-query';
+        // try {
+        $response = $this->guzzleClient->request($verb, $query, $options);
 
-        if (! empty($format)) {
-            $requestUri .= '&format=' . $format;
-        }
+        return $this->handleResponse($response);
+    }
 
-        return $requestUri;
+    /**
+     * Getter lastResponseCode
+     * @return int/null
+     */
+    public function getLastResponceCode()
+    {
+        return $this->lastResponseCode;
     }
 
     /**
-     * Make and return the request URI based on
-     * the passed query and the configured endpoint
+     * Handle response of request
      *
-     * @param  string $query
-     * @param  string $format
-     * @return string
+     * Set status code into property
+     * Check errors based on response status code
+     * Return the body of the request
+     *
+     * @param  Response $response
+     * @return Psr\Http\Message\StreamInterface
      */
-    private function makePostRequestUri($query, $format = null)
+    private function handleResponse(Response $response)
     {
-        $requestUri = $this->endpoint . '?query=' . $query;
+        $this->lastResponseCode = $response->getStatusCode();
+        // Virtuoso documentation states it only returns 200, 400, 500 status codes
+        // but apparently they mean 2xx, 4xx and 5xx
+        if ($response->getReasonPhrase() != 'OK' && $this->lastResponseCode > 299) {
+            $message = "Something went wrong while executing query.";
+            throw new \Exception($message);
+        }
+
+        if ($this->lastResponseCode >= '400' && $this->lastResponseCode <= '500') {
+            $message = "The SPARQL endpoint returned an error.";
```

Comment:
Does the $response contain the error message? If so, append it to the exception message, so we know what's going on.

Reply:
No: you get the error message from Guzzle what has nothing to do with sparql error.
The return is html from the website but not something we can log

Reply:
Too bad, but nothing we can do about it then.